wordpress-mobile/WordPress-iOS
# Jetpack Focus: Fix UI issues by removing the `WPTabBarController` dependency

Closes #19810
Part of #19812

## Description
In the simplified UI, we no longer display a WPTabBarController. However, many areas of the app depend on `WPTabBarController.sharedInstance` to perform some actions. In this PR, we completely remove this dependency and instead depend on the newly added `RootViewCoordinator`. By doing so, all issues in the simplified UI that were present in #19816 should be fixed.

The PR touches many parts of the app, but most changes are systematic.

## ⚠️ Disclaimer ⚠️ 
Some Jetpack features are still accessible from external sources, like deep linking, widgets, and the 3D touch menu. This will be addressed in a following PR.
Also, blogging reminders and weekly roundups are still not disabled.

## Testing Instructions
This PR touches many areas of the app. Therefore there aren't specific testing instructions. Instead, please smoke test the following areas, both in the normal app state and in the simplified UI state (if applicable):
- Login flows
- Sign-in flows
- Site creation flow
- Interacting with push notifications
- Spotlight search (From outside the app)
  - Opening app settings
  - Opening a published post
  - Opening a draft post
  - Opening Reader/Notifications
- 3D touch menu
- Reader
- What's new announcement (Check #18231 for testing steps)
- Creating a post/page/story from the floating button
- Creating a post/page from the floating button in the posts/pages list view
- Blogging prompts

To get to the normal app state, open the debug menu and disable all the Jetpack Features Removal flags. Then kill and relaunch the app.
To get to the simplified UI state, open the debug menu and enable "Jetpack Features Removal Phase Four". Then kill and relaunch the app.

## Regression Notes
1. Potential unintended areas of impact
Mentioned above.

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Manual testing

5. What automated tests I added (or what prevented me from doing so)
N/A

PR submission checklist:
